migrations/2024_04_24_01_edit_terms_policies.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+// HINT: you might want to use a `Flarum\Database\Migration` helper method for simplicity!
+// See https://docs.flarum.org/extend/models.html#migrations to learn more about migrations.
+return [
+    'up' => function (Builder $schema) {
+        $schema->table('fof_terms_policies', function (Blueprint $table) {
+            $table->json('additional_info');
```

Comment:
In this case, we can check the database version and, if it is too old, set the column as text. A column of type text will not cause problems because JSON is text and Eloquent will still cast the model attribute to an array in the same way.

@imorland Do you know how to execute a custom SQL query in Flarum? Unfortunately, the DB facade doesn't work.

```php
<?php

use Illuminate\Database\Schema\Blueprint;
use Illuminate\Database\Schema\Builder;
use Illuminate\Support\Facades\DB;
use Illuminate\Support\Str;

return [
    'up' => function (Builder $schema) {
        $schema->table('fof_terms_policies', function (Blueprint $table) {
            // JSON columns are supported on MariaDB 10.2+, MySQL 5.7+
            $requiredVersion = '5.7.8';
            $mySqlVersion = DB::select('SELECT VERSION() as version')[0]->version;

            if(Str::contains($mySqlVersion,'MariaDB')){
                $requiredVersion = '10.2.7';
            }

            if(version_compare($mySqlVersion, $requiredVersion,'>=')){
                $table->json('additional_info')->nullable();
            } else {
              $table->text('additional_info')->nullable();
            }
        });
    },
    'down' => function (Builder $schema) {
        $schema->table('fof_terms_policies', function (Blueprint $table) {
            $table->dropColumn('additional_info');
        });
    },
];
```

Reply:
I've submitted the change. The `additional_info` column will now be of type `LONGTEXT` instead of `JSON`. The effect will be the same, but there won't be native JSON validation in the database.